### PR TITLE
 Read and print message and filename in abort()

### PR DESCRIPTION
**Pull request by vgrichina**

### CODE REVIEW

**Comment by vgrichina** [comment]
This will be rebased after merging https://github.com/nearprotocol/nearcore/pull/193

File: core/wasm/src/runtime.rs [hunk 1]
```diff
@@ -379,16 +396,20 @@ impl<'a> Runtime<'a> {
...
 
-        debug!("msg: {:?} filename: {:?} line: {:?} col: {:?}", msg, filename, line, col);
+        debug!("abort with msg: {:?} filename: {:?} line: {:?} col: {:?}", msg, filename, line, col);
 
         Err(Error::AssertFailed)
     }
 
+    fn log(&self, args: &RuntimeArgs) -> Result<()> {
+        let msg_ptr: u32 = args.nth_checked(0)?;
+        println!("{}", self.read_string(msg_ptr).unwrap_or_else(|_| "log(): read_string failed".to_string()));
```
[the remark this answers is not in the document]
Comment (bowenwang1996) [reply core/wasm/src/runtime.rs]: I usually use logging in production code and println in test code.

File: core/wasm/src/runtime.rs [hunk 2]
```diff
@@ -6,6 +6,7 @@ use types::{RuntimeError as Error, ReturnData, RuntimeContext};
 
 use primitives::types::{AccountAlias, AccountId, PromiseId, ReceiptId, Balance, Mana, Gas};
 use std::collections::HashSet;
+use std::string::String;
```
[the remark this answers is not in the document]
Comment (vgrichina) [reply core/wasm/src/runtime.rs]: fixed in #193

File: core/wasm/src/runtime.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (vgrichina) [reply core/wasm/src/runtime.rs]: so if I use `debug!` here I won't see anything in devnet output when running a contract. Which makes whole presence of function pointless. 

I should be able to do something like:
```
DEBUG=1 cargo run --package=devnet 
```

then logging becomes usable

File: core/wasm/src/runtime.rs [hunk 3]
```diff
@@ -395,12 +395,10 @@ impl<'a> Runtime<'a> {
...
 
-        debug!("msg_ptr: {:?} filename_ptr: {:?} line: {:?} col: {:?}", msg_ptr, filename_ptr, line, col);
+        let msg = self.read_string(msg_ptr)?;
+        let filename = self.read_string(filename_ptr)?;
 
-        let msg = self.read_buffer_with_size(msg_ptr, 32)?;
-        let filename = self.read_buffer_with_size(filename_ptr, 32)?;
-
-        debug!("msg: {:?} filename: {:?} line: {:?} col: {:?}", msg, filename, line, col);
+        debug!("abort with msg: {:?} filename: {:?} line: {:?} col: {:?}", msg, filename, line, col);
```
Comment (azban) [inline core/wasm/src/runtime.rs]: you'll want to pass `target: wasm` so you can enable debug logs

Comment (azban) [reply core/wasm/src/runtime.rs]: like so: https://github.com/nearprotocol/nearcore/blob/master/node/runtime/src/lib.rs#L687

**Pull request merged**